examples/src/main/java/com/jsyn/examples/PlayFunction.java

```diff
@@ -17,59 +17,58 @@
 package com.jsyn.examples;
 
 import com.jsyn.JSyn;
-import com.jsyn.Synthesizer;
 import com.jsyn.data.Function;
 import com.jsyn.unitgen.FunctionOscillator;
 import com.jsyn.unitgen.LineOut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Play a tone using a FunctionOscillator.
- * 
+ *
  * @author Phil Burk (C) 2010 Mobileer Inc
  */
 public class PlayFunction {
-    Synthesizer synth;
-    FunctionOscillator osc;
-    LineOut lineOut;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PlayFunction.class);
 
     private void test() {
         // Create a context for the synthesizer.
-        synth = JSyn.createSynthesizer();
+        var synth = JSyn.createSynthesizer();
 
         // Start synthesizer using default stereo output at 44100 Hz.
         synth.start();
 
         // Add a FunctionOscillator
-        synth.add(osc = new FunctionOscillator());
+        var oscillator = new FunctionOscillator();
+        synth.add(oscillator);
 
         // Define a function that gives the shape of the waveform.
-        Function func = new Function() {
-            @Override
-            public double evaluate(double input) {
-                // Input ranges from -1.0 to 1.0
-                double s = Math.sin(input * Math.PI * 2.0);
-                double cubed = s * s * s;
-                return cubed;
-            }
+        Function func = input -> {
```

Comment:
Does this work? Does FunctionOscillator still work the same or does it break old code?

Reply:
It works the exact same, if there's only one method in the interface you can override it with its parameters.